src/Gui/BitmapFactory.cpp

```diff
@@ -325,15 +326,26 @@ QPixmap BitmapFactoryInst::pixmapFromSvg(const char* name, const QSize& size) co
         QFile file(iconPath);
         if (file.open(QFile::ReadOnly | QFile::Text)) {
             QByteArray content = file.readAll();
-            icon = pixmapFromSvg(content, size);
+            icon = pixmapFromSvg(content, size, colorMapping);
         }
     }
 
     return icon;
 }
 
-QPixmap BitmapFactoryInst::pixmapFromSvg(const QByteArray& contents, const QSize& size) const
+QPixmap BitmapFactoryInst::pixmapFromSvg(const QByteArray& originalContents, const QSizeF& size,
+    const std::map<unsigned long, unsigned long> colorMapping) const
 {
+    QString stringContents = QString::fromUtf8(originalContents);
+    for ( const auto &colorToColor : colorMapping ) {
+        ulong fromColor = colorToColor.first;
+        ulong toColor = colorToColor.second;
+        QString fromColorString = QString::fromLatin1(":#%1;").arg(fromColor, (int)6, (int)16,  QChar::fromLatin1('0'));

```

Comment:
The C-style cast of (int) is superfluous, isn't it? And as a general note: In new code we should avoid C-style casts completely -- even for basic types.

Reply:
done